showyourwork/workflow/resources/styles/build.tex

```diff
@@ -43,133 +52,73 @@
+\definecolor{sywStampRed}{rgb}{0.80,0.45,0.45}
+\definecolor{sywOrange}{rgb}{1.0,0.65,0.0}
+\definecolor{sywYellow}{rgb}{1.0,0.88,0.30}
+\definecolor{sywBlue}{rgb}{0.12,0.47,0.71}
+\definecolor{sywGreen}{rgb}{0.13,0.53,0.23}
 
-% MNRAS hack for displaying showyourwork logo
-\@ifclassloaded{mnras}{
-  \ifx\CUP@mtlplain@loaded\undefined
-    \def\@printed{Preprint \tod@y\
-      \qquad\qquad\qquad Compiled using \showyourwork/MNRAS \LaTeX\ style file v\@version
-      \ifcsname gitHeader\endcsname
-        \,(\gitHeader)
-      \fi
-    }
-  \else
-    \def\@printed{}
-  \fi
-}{}
-
-% A&A hack for displaying showyourwork logo + abstract links
-\@ifclassloaded{aa}{
-  \renewcommand*\aa@manuscriptname{%
-    \LaTeX
-    \hspace{\stretch{1}}%
-    \smash{\stackunder[0.5em]{\copyright ESO \the\year}{%
-        \href{\syw@url/tree/\syw@sha/}{%
-          \color{sywBlue}%
-          \faGithub%
-        }%
-        \ifOnGithubActions
-          \hspace{3pt}%
-          \href{\syw@url/actions/runs/\syw@runid/}{%
-            \color{sywGreen}%
-            \faCheckCircle%
-          }%
-        \fi
-      }
-    }
+% Generate the stamp on the first page
+\ifAddStamp
+  % Stamp position hacks (class-specific, as needed)
+  \@ifclassloaded{aastex631}{
+    \newcommand{\syw@stampXOffset}{-0.60}
+    \newcommand{\syw@stampYOffset}{0.00}
+  }{
+    \newcommand{\syw@stampXOffset}{0.00}
+    \newcommand{\syw@stampYOffset}{0.00}
   }
-}{}
-
-% Annual Reviews hack for displaying showyourwork logo
-\@ifclassloaded{ar-1col}{
-  \let\oldjournalinfo\journalinfo
-  \def\journalinfo{%
-    \oldjournalinfo%
-    \vskip6\p@{\jinfofont%
-      Compiled using\begin{minipage}{7.3em}\includegraphics[width=7.3em]{showyourwork-logo.pdf}\end{minipage}\endgraf%
-      \ifcsname gitHeader\endcsname
-        \,(\gitHeader)\endgraf
-      \fi
+  % Define the stamp
+  \newcommand{\syw@stamp}{%
+    \pgfmathsetmacro{\radius}{\syw@stampSize*0.66}%
+    \pgfmathsetmacro{\sx}{\syw@stampX + \syw@stampXOffset + \syw@stampSize/2}%
+    \pgfmathsetmacro{\sy}{\syw@stampY + \syw@stampYOffset + \syw@stampSize/2}%
+    \pgfmathsetmacro{\ax}{\radius*sin(\syw@stampAngle)}%
+    \pgfmathsetmacro{\ay}{-\radius*cos(\syw@stampAngle)}%
+    \href{\syw@url/tree/\syw@sha/}{%
+      \begin{tikzpicture}[remember picture,overlay]
+        \node (STAMP) at ($(current page.north east) - (\sx in, \sy in)$)
+          [rotate=\syw@stampAngle]
+          {\includegraphics[width=\syw@stampSize in]{showyourwork-stamp.pdf}};
+        \draw[draw=none,
+          postaction={decorate},
+          decoration={
+            text along path,
+            text={|\bfseries\ttfamily\syw@stampFontSize|\syw@shortUrl},
+            text align=center,
+            text color=sywStampRed
+          }]
+          ($(STAMP.center) - (\ax in, \ay in)$)
+          arc
+          [start angle={\syw@stampAngle+90},
+          end angle={\syw@stampAngle+450},
+          radius=\radius in];
+      \end{tikzpicture}%
```

Comment:
I also wonder if you'd like to put the SYW version information somewhere in the stamp. e.g., perhaps in the future, the stamp style changes. I wouldn't want people to think I am using a knockoff SYW, trying to pretend my work is reproducible :P - showing the version number would explain an older stamp style.

(Maybe this is too much info/too much work - fine by me!)

Reply:
This is a good idea. It's not too much work -- should be easy with TikZ. The only issue is that the version can be a number, a tag, a branch, a commit, a path, or a fork. Some of those won't fit in the stamp. Perhaps we could display the version as long as it's a pip-installable version, and (maybe) just ``dev`` otherwise?

Reply:
Working on this right now.

Reply:
Done. The version now appears inside the stamp if it's a pip-installable version; otherwise it displays ``dev``.